### PR TITLE
feat: export sider context from layout

### PR DESCRIPTION
**Pull request by PeachScript**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

None

### 💡 Background and solution

Export `SiderContext` from `Layout`, to avoid multi-instance problem when using `ProLayout` and external `antd` at the same time, ref: https://github.com/ant-design/pro-components/blob/8b451c6e255fecced11e7d7b0be4aad2bfa0a09c/packages/layout/src/components/SiderMenu/SiderMenu.tsx#L5

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Layout export SiderContext |
| 🇨🇳 Chinese | Layout 组件导出 SideContext  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 1cc16ab</samp>

Add `SiderContext` to `Layout` component to support responsive `Sider` and its children. This context exposes `Sider`'s collapsed state and width to its descendants.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 1cc16ab</samp>

*  Expose `SiderContext` as a static property of `Layout` component to provide `Sider` information to children ([link](https://github.com/ant-design/ant-design/pull/45916/files?diff=unified&w=0#diff-87da5c09eb7577d425c8ee19da79197d0b702d7ba6a47d2204d4170e2c584b81L2-R2), [link](https://github.com/ant-design/ant-design/pull/45916/files?diff=unified&w=0#diff-87da5c09eb7577d425c8ee19da79197d0b702d7ba6a47d2204d4170e2c584b81R14), [link](https://github.com/ant-design/ant-design/pull/45916/files?diff=unified&w=0#diff-87da5c09eb7577d425c8ee19da79197d0b702d7ba6a47d2204d4170e2c584b81R23))

### CODE REVIEW

File: components/layout/index.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ type CompoundedComponent = InternalLayoutType & {
   Footer: typeof Footer;
   Content: typeof Content;
   Sider: typeof Sider;
+  SiderContext: typeof SiderContext;
```
Comment (MadCcc) [inline components/layout/index.ts:14]: 加个 internal？

**Review by killagu — APPROVED** [review]
LGTM

File: components/layout/index.ts (hunk 1 above)
Comment (PeachScript) [reply components/layout/index.ts:14]: 叫 `InternalSiderContext` 还是 `__internalSiderContext`

Comment (MadCcc) [reply components/layout/index.ts:14]: tsdoc 打个 private 的 flag 吧，和 PurePanel 一样叫 `_InternalSiderContext` 吧

**Review by MadCcc — APPROVED** [review]

**Pull request merged**